knowledge-base/grid-search-numeric-fields.md

```diff
@@ -0,0 +1,313 @@
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+How to allow users search the Blazor Grid in numeric, date and enum fields?
+
+How to use the toolbar GridSearchBox to search values in number columns?
+
+How to search integers or float numbers from the Grid toolbar?
+
+How to make the Grid SearchBox search with multiple data types?
+
+
+## Solution
+
+The Blazor Grid search works with `string` data types, because it uses a `contains` string filter operator.
+
+It is possible to search by other data types with some additional coding that will use non-string filter operators.
+
+The following sections discuss how to implement the desired behavior, what are the limitations, and provide an example.
+
+
+## Prerequisites
+
+The solution below requires familiarity with:
+
+* [How to use Grid filter descriptors]({%slug components/grid/filtering%}#filter-descriptors)
+* [How to change the Grid state]({%slug grid-state%})
+* [How to invoke Razor component events from a child component](https://docs.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-6.0#eventcallback).
+
+
+## Steps
+
+Here is an overview of the major steps in the example:
+
+1. Add a search textbox to the [Grid toolbar]({%slug components/grid/features/toolbar%}) or outside the Grid. Do not use the default [`GridSearchBox` component]({%slug grid-searchbox%}), as it searches inside string only.
+1. Handle the `oninput` event of the search textbox. Alternatively, use a search button with an `onclick` handler.
+1. Use the event handler to:
+    * Create a new `CompositeFilterDescriptor`.
+    * Iterate all searchable fields of the Grid.
+    * For each searchable field, parse the search string and convert it the field's data type. Create a child `FilterDescriptor` object, depending on the data field type and the parsed search value.
+1. Apply the filter descriptors via the Grid state.
+
+
+## How it works
+
+* A search value `"8"` will return:
+    * *string* results that *contain* the string "8"
+    * *numeric* results that are *equal to* 8
+    * *enums* results that are *equal to* `int` 8
+* Date search values *must fully match* date values in the Grid to return results. The example also shows how to search by year only.
+* If the search value cannot be parsed to a certain type, it won't add filters for columns of this type. For example, a search value of `"8 asd"` won't return any numeric, date and boolean results
+
+>tip It is possible to achieve different behavior, or to work around the above limitations, by using different business logic in the filter descriptor creation.
+
+> The example is incompatible with [regular Grid filtering]({%slug components/grid/filtering%}), because the `SetGridSearchBoxFilters` method always removes all existing filter descriptors: `state.FilterDescriptors.Clear()`. The Grid and its API does not distinguish programmatic filters and filters set by the user.
```

Comment:
```suggestion
> The example is incompatible with [regular Grid filtering]({%slug components/grid/filtering%}). The `SetGridSearchBoxFilters` method always removes all existing filter descriptors, because the Grid and its API does not distinguish programmatic filters and filters set by the user. In addition, the total number of programmatic and user filters will exceed the abilities of the Grid filtering UI.
```